src/PackageCompiler.jl

```diff
@@ -507,7 +507,14 @@ function create_sysimage(packages::Union{Symbol, Vector{String}, Vector{Symbol}}
     # Need to ensure that Pkg can find the source location of the packages:
     copy!(LOAD_PATH, [project])
     try
-        Pkg.instantiate(ctx, verbose=true, allow_autoprecomp = true)
+        Pkg.instantiate(ctx, verbose=true, allow_autoprecomp = false)
+        if :warn_loaded in Base.kwarg_decl(first(methods(Pkg.precompile, Tuple{Pkg.Types.Context})))
```

Comment:
Hm, I also like feature detection but that feels almost feels too much :P. Maybe best to wait for a julia version and just do a regular `VERSION` check?

Reply:
the `first(...)` does make this feel a bit too brittle